### ISSUE TITLE
mypy doesn't catch inheritance of outer class

### ISSUE THREAD

**Issue by injust** [python/mypy#18393]
**Bug Report**

Inheriting from an outer class raises `NameError` at runtime, but mypy doesn't catch this.

**To Reproduce**

```python
class Foo:
    class Bar(Foo):
        pass
```

**Expected Behavior**

**Actual Behavior**

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 1.14.0
- Python version used: 3.13
